### PR TITLE
feat(ci) adjust so running the functional test can be ran isolated

### PR DESCRIPTION
**Pull request by hutchic**
Adjust how the tests are ran so they can more easily be ran in isolation so other Kong CI/CD can use the functional tests

### CODE REVIEW

File: test/test_smoke.tavern.yaml [hunk 1]
```diff
@@ -80,8 +80,6 @@ stages:
         url: http://randomapiname.com/request
       headers:
         Server: openresty/1.13.6.2
-        Via: kong/1.0.2
+        Via: kong/1.0.3
```
[the remark this answers is not in the document]
Comment (hutchic) [reply test/test_smoke.tavern.yaml]: Yup it should be / will be 👍

File: .travis.yml [hunk 2]
```diff
@@ -32,7 +32,8 @@ matrix:
     - env: PACKAGE_TYPE=deb RESTY_IMAGE_BASE=debian RESTY_IMAGE_TAG=testing
 
 install:
-  - git clone --single-branch -b next https://github.com/Kong/kong.git kong
+  - git clone https://github.com/Kong/kong.git kong
+  - cd kong && git reset --hard 1fc45e5
```
Comment (hutchic) [inline .travis.yml]: remove once https://github.com/Kong/kong/pull/4279 is merged

File: test/test_smoke.tavern.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (hutchic) [reply test/test_smoke.tavern.yaml]: resolved in 21df471

File: Makefile [hunk 3]
```diff
@@ -19,10 +21,16 @@ endif
 KONG_PACKAGE_NAME?="kong-community-edition"
 KONG_CONFLICTS?="kong-enterprise-edition"
 KONG_LICENSE?="ASL 2.0"
-KONG_SOURCE_LOCATION?="$$PWD/../kong/"
-KONG_VERSION?="0.0.0"
 PRIVATE_REPOSITORY?=true
 KONG_TEST_CONTAINER_NAME?=localhost:5000/kong
+KONG_SOURCE_LOCATION?="$$PWD/../kong/"
+
+KONG_VERSION?=`grep tag $(KONG_SOURCE_LOCATION)/kong-*.rockspec | awk -F" " '{print $$3}' | sed 's/"//g'`
```
Comment (hishamhm) [inline Makefile]: the value of the `tag` field there is always the same as the `*` bit in `kong-*.rockspec`, but extracting it from the filename is more reliable than awk+sed'ding out of the contents.
```suggestion
KONG_VERSION?=`echo $(KONG_SOURCE_LOCATION)/kong-*.rockspec | sed 's,.*/,,' | cut -d- -f2`
```

File: .travis.yml (hunk 2 above)
Comment (hishamhm) [reply .travis.yml]: ahh, missed this note, sorry

File: .travis.yml [hunk 4]
```diff
@@ -32,7 +32,8 @@ matrix:
     - env: PACKAGE_TYPE=deb RESTY_IMAGE_BASE=debian RESTY_IMAGE_TAG=testing
 
 install:
-  - git clone --single-branch -b next https://github.com/Kong/kong.git kong
+  - git clone https://github.com/Kong/kong.git kong
+  - cd kong && git reset --hard 1fc45e5 && cd ../
```
[the remark this answers is not in the document]
Comment (hishamhm) [reply .travis.yml]: nvm, saw note above

File: Makefile [hunk 5]
```diff
@@ -19,10 +21,16 @@ endif
 KONG_PACKAGE_NAME?="kong-community-edition"
 KONG_CONFLICTS?="kong-enterprise-edition"
 KONG_LICENSE?="ASL 2.0"
-KONG_SOURCE_LOCATION?="$$PWD/../kong/"
-KONG_VERSION?="0.0.0"
 PRIVATE_REPOSITORY?=true
 KONG_TEST_CONTAINER_NAME?=localhost:5000/kong
+KONG_SOURCE_LOCATION?="$$PWD/../kong/"
+
```
Comment (jeveleth) [inline Makefile]: Remove extra line break?

File: README.md [hunk 6]
```diff
@@ -60,10 +60,38 @@ make build_test_container
 - Minikube
 - Helm
 
+Which assuming Linux can be installed via `make setup_tests`
```
Comment (jeveleth) [inline README.md]: I don't understand this sentence.

Comment (hutchic) [reply README.md]: Would this make more sense
> The above requirements can be installed via `make setup_tests` if running Linux

Comment (hutchic) [reply README.md]: resolved in bdf093e

File: Makefile [hunk 7]
```diff
@@ -19,10 +21,15 @@ endif
 KONG_PACKAGE_NAME?="kong-community-edition"
 KONG_CONFLICTS?="kong-enterprise-edition"
 KONG_LICENSE?="ASL 2.0"
-KONG_SOURCE_LOCATION?="$$PWD/../kong/"
-KONG_VERSION?="0.0.0"
 PRIVATE_REPOSITORY?=true
 KONG_TEST_CONTAINER_NAME?=localhost:5000/kong
+KONG_SOURCE_LOCATION?="$$PWD/../kong/"
+KONG_VERSION?=`echo $(KONG_SOURCE_LOCATION)/kong-*.rockspec | sed 's,.*/,,' | cut -d- -f2`
```
Comment (jeveleth) [inline Makefile]: Any chance this line and [this line](https://github.com/Kong/kong/blob/3a8abf8591e9f73228732150920e3cd18b2cce1b/Makefile#L22) could be DRYed up into a ENVVAR in the `.requirements` file?

Comment (hutchic) [reply Makefile]: not at this time. Right now there's too much plumbing in other places that expect the rockspec to be versioned. Future us should try to collate to one source of truth 👍

File: .travis.yml (hunk 2 above)
Comment (kikito) [reply .travis.yml]: Kong/kong#4279 is merged so maybe this can be removed now.

Comment (hutchic) [reply .travis.yml]: 4b1d474 just waiting on travis-ci now

**Pull request merged**